client/src/main/java/io/pravega/client/stream/EventStreamReader.java

```diff
@@ -71,4 +71,12 @@
      */
     @Override
     void close();
+
+    /**
+     * Close the reader. No further actions may be performed. If this reader is part of a {@link ReaderGroup}, this will
+     * automatically invoke {@link ReaderGroup#readerOffline(String, Position)} with the supplied position.
+     *
+     * @param position {@link Position} to use while reporting readerOffline on the {@link ReaderGroup}.
+     */
+    void close(Position position);
```

Comment:
I'm 👎 for this change. This is introducing a close call to a public facing API that has an argument while this class extends `AutoCloseable`. This is far from being elegant. If we need to remove the reader at a given position, can't we invoke `readerOffline`, which is a call that exists precisely for that purpose?

Reply:
If readerOffline fails, we still want to close the reader (so that any resources taken by the object are not leaked). but since reader.close explicitly sets the position which we do not want, we want to merge the two methods into one so that we tell the close method which position to use while closing.

Reply:
The alternative is to have a flag in the reader which can disable automatic update of position during close. So even if readergroup.readerOffline fails, we dont have to worry about reader updating a position that we never intended it to.

Reply:
I'm not sure readerOffline failing should be the concern. 
The larger case here is one of ergonomics. Calling close exits gracefully at the current position. 
readerOffline notifies the reader group the reader has already closed at a particular position. But it does seem weird to invoke readerOffline followed by close on the client when it wants to exit from a different position. Though it does work. 

That being said, autoclosable is a common enough of a thing that it might end up invoking close() after close(pos) just do to how code is structured. (We can tolerate that, but it's still weird.)

Reply:
@tkaitchuck and I had an offline discussion about this and I want to reflect that conversation here along with the suggestion.

- We understand that the flow of calling `readerOffline` to stop an event reader at a given `Position` is odd because we need to instantiate a different object (a `ReaderGroup`) to be able to call it.
- Based on the previous point, we really need a call in `EventStreamReader` that stops at a given position, which is essentially the approach proposed here.
- If we are to create a new method, then we have two options:
   1- Have an additional call to set the position and then call close.
   2- Have an additional call that replaces the call to close. 
- Calling the method `close` has the problem of conflicting with the `close` from `AutoCloseable`. We started discussing options to reflect what it is that we are trying to do. We somehow are trying to close "ungracefully" because we are not closing at the last position. It is not ungraceful in the sense that reader is dropping it on the floor, but in the sense that it is not acknowledging that it has processed everything it has read. Consequently, we have gone through few options of names: `die`, `shutdown`, `stop`. To differentiate from the `close` of `AutoCloseable` but keep the semantics intact, we suggest to name it `closeAt(Position)`.

That's a minimal change and should unblock this PR.

Reply:
Another comment I forgot to add is that the javadoc should capture precisely why we need this call and in which cases the application should consider using it.